### PR TITLE
add ST's vl53l1x API

### PR DESCRIPTION
**Pull request by chris-makaio**
This adds ST's API for VL53L1X ToF sensor.
See UM2356 for driver description

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,6 +17,21 @@ if(CONFIG_HAS_STLIB)
     zephyr_sources(sensor/vl53l0x/api/core/src/vl53l0x_api_core.c)
     zephyr_sources(sensor/vl53l0x/api/core/src/vl53l0x_api_strings.c)
   endif()
+  if(CONFIG_VL53L1X)
+    zephyr_include_directories(
+            sensor/VL53L1X/api/core/inc
```
Comment (erwango) [inline CMakeLists.txt]: use lower case

**Comment by chris-makaio** [comment]
Thanks for the feedback. I mostly cloned the VL53L0X' README, I hope it's okay

**Comment by chris-makaio** [comment]
@avisconti alright, done

**Comment by avisconti** [comment]
@chris-makaio 
Just an additional question.
Can you confirm that the code you submitted is exactly as the one downloaded from ST?
I just checked one file, so I would like you to confirm that ALL the files are exactly the same as provided in the ST package.

**Comment by chris-makaio** [comment]
I just confirmed with "Beyond Compare" - all files are exactly the same. Only difference is that I've split headers and sources to src/inc, so, it's just a change in structure, not in files

**Comment by avisconti** [comment]
> I just confirmed with "Beyond Compare" - all files are exactly the same. Only difference is that I've split headers and sources to src/inc, so, it's just a change in structure, not in files

Yes, I think structure may change. In case of stmemsc I kept also the structure as same, so that it would be easier to upgrade to new module version.

**Comment by chris-makaio** [comment]
Is something left to do for me?

**Comment by avisconti** [comment]
> Is something left to do for me?

Did you address @erwango requests?

**Comment by chris-makaio** [comment]
Yes I did

**Comment by avisconti** [comment]
@erwango 
Erwan, can you possibly take a look again at this PR?

**Comment by chris-makaio** [comment]
but this HAL shouldn't need a license update... should it?

**Pull request merged**